functions/sudope.fish

```diff
@@ -1,16 +1,24 @@
-function sudope -d "Quickly toggle sudo prefix"
+function sudope -d "Quickly toggle super user command prefix"
 
   # Save the current command line and cursor position.
   set -l command_buffer (commandline)
   set -l cursor_position (commandline -C)
 
+  # If no custom super user command defaults to sudo
+  if test -z "$sudope_command"
+    set sudope_command 'sudo'
+  end
+
+  # Save su command length
+  set -l sudope_command_length (string length -- "$sudope_command")
+
   # If the command line is empty, pull the last command from history.
   if test -z "$command_buffer"
     set command_buffer $history[1]
   end
 
   # Parse the command line (first line only).
-  set -l command_parts (string match -ir '^(\s*)(sudo(\s+|$))?(.*)' $command_buffer[1])
+  set -l command_parts (string match -ir "^(\s*)($sudope_command(\s+|\$))?(.*)" $command_buffer[1])
```

Comment:
we should probably regex escape $sudope_command (`string escape --style=regex -- "$sudope_command"` and store it somewhere?)

Reply:
I didn't know the `string` builtin. Cool stuff, thanks! Can you explain to me its role here? `$sudope_command` it's supposed to be a shell command (eg. `sudo`, `doas`, et cetera). Are you thinking about something like `mkfs.ext2`, that in fact it's a problem here?

Reply:
yeah, `.` was the thing i was most interested in. but mostly it's defensive programming. interpolating user-provided values into a regex introduces opportunities for things to go wrong, and it's unnecessary when we have a way to escape them right here :)

Reply:
Perfect, I'm with you :)

Reply:
Fixed in 11b1fca97739289e138e4d0e8c7c5cec2a24609f.